packages/flutter_tools/lib/src/dart/pub.dart

```diff
@@ -480,4 +499,18 @@ class _DefaultPub implements Pub {
       await savePackageConfig(newPackageConfig, packageConfigFile.parent.parent);
     }
   }
+
+  // Subset the package config file to only the parts that are relevant for
+  // rerunning the dart compiler.
+  String _computePackageConfigSubset(PackageConfig packageConfig, FileSystem fileSystem) {
+    final StringBuffer buffer = StringBuffer();
+    for (final Package package in packageConfig.packages) {
+      buffer.write(package.name);
+      buffer.write(package.languageVersion);
+      buffer.write(package.root);
+      buffer.write(package.packageUriRoot);
```

Comment:
Should these be separated by newlines?  Doesn't matter now, but it'll be hard to parse back as a single line if we ever need to.

Reply:
Done